### ISSUE TITLE
when using on_delete = true with additional persistent disks, deleting the disk fails

### ISSUE THREAD

**Issue by shpsec-dennis** [hashicorp/terraform#5942]
Hi,

Terraform seems to try delete persistent disks created by it even if they were already deleted by the Google API, because on_delete = True (the terraform default afaict).

This is what the output looks like:

```
$ terraform destroy
google_compute_disk.diskname-disks.1: Destroying...
google_compute_disk.diskname-disks.0: Destroying...
Error applying plan:

2 error(s) occurred:

* google_compute_disk.diskname-disks.0: Error deleting disk: googleapi: Error 404: The resource 'projects/projectname/zones/us-central1-f/disks/diskname-disk-0' was not found, notFound
* google_compute_disk.diskname-disks.1: Error deleting disk: googleapi: Error 404: The resource 'projects/projectname/zones/us-central1-f/disks/diskname-disk-1' was not found, notFound
```

Running `$ terraform destroy` seems to fix this, because it just does a refresh of the state.

Since terraform knows (or should) that it created the disk with on_destroy, delete the disk logic it might refer to it when doing the destroy, or do more than one refresh during a destroy?

Thank you for your time,

Dennis O.

**Comment by clstokes** [comment]
This is a bigger issue than just the error message. This could lead to unintentional data loss.

This is because Google is deleting the attached disk separately from Terraform managing it. This has several consequences - it's annoying (best case), it breaks Terraform's resource graph for taints, recreating the instance, etc, and could very likely could delete my disk when I didn't want it too (worst case scenario).

I think the 404 error should be ignored. The same error is ignored when I delete the `google_compute_disk` resource in the Google cloud console and then do a terraform destroy.

**More importantly**, I think `auto_delete = true` and specifying an existing disk on an instance (using `disk` argument on the `disk` sub-resource) should conflict and Terraform should not allow this configuration to prevent this potentially dangerous configuration that could lead to data loss.

Terraform config and log: https://gist.github.com/clstokes/f627afd386dd0b1d908e6259f9b87fe5

**Comment by phinze** [comment]
Good points @clstokes - @lwander @evandbrown does the above make sense to you as reasonable changes to make?

**Comment by lwander** [comment]
I agree @clstokes, this config should throw an error on `apply`.

**Comment by dvianello** [comment]
Fully agree with @clstokes, the way it works now it's dangerous!

**Comment by evandbrown** [comment]
Sorry for weighing in late here.

@clstokes, I'm a little worried about disabling `auto_delete` all together for existing disks. What do you think about changing the default value of `auto_delete` to false? This is the default behavior in the web console, and fits nicely with what most people would probably expect from the default value of a bool.

Is there precedent for changing default values in Terraform? I could imagine a user expecting `auto_delete=true` being surprised if this changed, but we'd clearly document it.

I'll get a fix in for correct instance deletion with `auto_delete=true` disks shortly, and tack on the other change after y'all have weighed in.

Thanks!!

**Comment by jen20** [comment]
This should be fixed by #6689.

**Comment by clstokes** [comment]
@evandbrown I like the idea of changing the default `auto_delete` to `false` and think this still needs to be done. @phinze @jen20 can weigh in on _how_ to do this change, but it seems there's precedent for things like this in our changelog with a backwards incompatibility note.
